crates/shared/src/state.rs

```diff
@@ -159,6 +159,24 @@ where
             <Types::BlockPayload as BlockPayload<Types>>::from_bytes(encoded_txns, metadata);
         let txn_commitments = block_payload.transaction_commitments(metadata);
 
+        let views_since_nonempty_block = if txn_commitments.is_empty() {
```

Comment:
This is used by legacy builder to decide whether we want to build empty block to prioritize finalization.